apps/server/src/modules/school/service/school.service.spec.ts

```diff
@@ -190,74 +203,106 @@ describe('SchoolService', () => {
+				expect(resp.oauthMigrationPossible).toEqual(testDO.oauthMigrationPossible);
+				expect(resp.oauthMigrationMandatory).toEqual(testDO.oauthMigrationMandatory);
+				expect(resp.oauthMigrationFinished).toEqual(testDO.oauthMigrationFinished);
+				expect(schoolRepo.findById).toHaveBeenCalledWith(testId);
+				expect(schoolRepo.save).toHaveBeenCalledWith(testDO);
+			});
 		});
 	});
 
-	describe('getMigration', () => {
+	describe('getMigration is called', () => {
 		let testId: string;
 		let testDO: SchoolDO;
 		beforeEach(() => {
 			testId = 'migration';
 			testDO = new SchoolDO({
 				id: testId,
 				name: 'testDO',
-				oauthMigrationPossible: true,
-				oauthMigrationMandatory: true,
+				oauthMigrationPossible: new Date(),
+				oauthMigrationMandatory: new Date(),
+				oauthMigrationFinished: new Date(),
+				officialSchoolNumber: '1337',
 			});
 			schoolRepo.findById.mockResolvedValue(testDO);
 		});
-		it('it should get the migrationflags', async () => {
-			const resp: MigrationResponse = await schoolService.getMigration(testId);
-			expect(resp.oauthMigrationPossible).toBeTruthy();
-			expect(resp.oauthMigrationMandatory).toBeTruthy();
-			expect(schoolRepo.findById).toHaveBeenCalledWith(testId);
+
+		describe('when migrationflags and officialSchoolNumber are set and schoolId is given', () => {
+			it('it should get the migrationflags', async () => {
+				const resp: MigrationResponse = await schoolService.getMigration(testId);
+
+				expect(resp.oauthMigrationPossible).toEqual(testDO.oauthMigrationPossible);
+				expect(resp.oauthMigrationMandatory).toEqual(testDO.oauthMigrationMandatory);
+				expect(resp.oauthMigrationFinished).toEqual(testDO.oauthMigrationFinished);
+				expect(resp.enableMigrationStart).toBeTruthy();
+				expect(schoolRepo.findById).toHaveBeenCalledWith(testId);
+			});
 		});
-		it('it should get the migrationflags when not set', async () => {
-			testDO.oauthMigrationPossible = undefined;
-			testDO.oauthMigrationMandatory = undefined;
-			const resp: MigrationResponse = await schoolService.getMigration(testId);
-			expect(resp.oauthMigrationPossible).toBeFalsy();
-			expect(resp.oauthMigrationMandatory).toBeFalsy();
-			expect(schoolRepo.findById).toHaveBeenCalledWith(testId);
+
+		describe('when migrationfalgs and officialSchoolNumber are not set and schoolId is given', () => {
+			it('it should get the migrationflags when not set', async () => {
+				testDO.oauthMigrationPossible = undefined;
+				testDO.oauthMigrationMandatory = undefined;
+				testDO.oauthMigrationFinished = undefined;
+				testDO.officialSchoolNumber = undefined;
+				const resp: MigrationResponse = await schoolService.getMigration(testId);
+				expect(resp.oauthMigrationPossible).toBeUndefined();
+				expect(resp.oauthMigrationMandatory).toBeUndefined();
+				expect(resp.oauthMigrationFinished).toBeUndefined();
+				expect(resp.enableMigrationStart).toBeFalsy();
+				expect(schoolRepo.findById).toHaveBeenCalledWith(testId);
```

Comment:
The findById expection should be tested in a seperated testcase

Reply:
!